src/message.rs

```diff
@@ -77,17 +77,55 @@ impl MsgId {
     }
 
     /// Returns true if the message needs to be moved from `folder`.
-    pub async fn needs_move(self, context: &Context, folder: &str) -> Result<bool, Error> {
-        if !context.get_config_bool(Config::MvboxMove).await {
-            return Ok(false);
-        }
-
+    /// If yes, returns `ConfiguredInboxFolder` or `ConfiguredMvboxFolder`, depending
+    /// on where the message should be moved
+    pub async fn needs_move(
+        self,
+        context: &Context,
+        folder: &str,
+    ) -> Result<Option<Config>, Error> {
+        use Config::*;
         if context.is_mvbox(folder).await {
-            return Ok(false);
+            return Ok(None);
         }
 
         let msg = Message::load_from_db(context, self).await?;
```

Comment:
When running the move job, we already have the Message struct and create it again here, so that I thought about adding an Option<Message> where the caller can put the Message object. But then it's "only" in the background (not when fetching messages) and the code started looking ugly when I started implementing this.